### ISSUE TITLE
Ejabberd not starting with custom module - [critical] <0.303.0>@gen_mod:start_module:206 Problem starting the module mod_abc

### ISSUE THREAD

**Issue by coolfeature** [processone/ejabberd#1654]
> What version of ejabberd are you using?

branch master

> What operating system (version) are you using?

linux

> How did you install ejabberd (source, package, distribution)?

source

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

ejabberd is not starting up with a custom module enabled

```
[warning] <0.303.0>@gen_mod:sort_modules:149 module 'mod_mam' is recommended for module 'mod_muc' but is not found in the config
2017-04-04 16:56:47.780 [critical] <0.303.0>@gen_mod:start_module:206 Problem starting the module mod_abc for host localhost
 options: []
 error: undef
[{mod_abc ,start,[<<"localhost">>,[]],[]},
 {gen_mod,start_module,3,[{file,"src/gen_mod.erl"},{line,194}]},
 {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
 {gen_mod,start_link,0,[{file,"src/gen_mod.erl"},{line,76}]},
 {supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,365}]},
 {supervisor,start_children,3,[{file,"supervisor.erl"},{line,348}]},
 {supervisor,init_children,2,[{file,"supervisor.erl"},{line,314}]},
 {gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]}]
2017-04-04 16:56:47.780 [critical] <0.303.0>@gen_mod:maybe_halt_ejabberd:285 ejabberd initialization was aborted because a module start failed.
```

Module implements gen_mod but does not have gen_server behaviour. 

sbin/ejabberdctl module_check mod_abc returns ok

**Comment by zinid** [comment]
`undef` means either there is no `mod_abc.beam` file or function `start/2` is not exported/existed.

**Comment by coolfeature** [comment]
start/2 exists otherwise module check would not work? And if the module checking works then ejabberd should be finding the .beam file too?

**Comment by zinid** [comment]
If you don't put it in `ejabberd.yml` and load it by hands (via `gen_mod:start_module/3`), do you have this error?

**Comment by cromain** [comment]
At start, ext_mod prepends every $CONTRIB_MODULES_PATH/*/ebin to the code path.
ext_mod must be started before gen_mod. maybe that's not the case cause of 67d6ca9f1

**Comment by cromain** [comment]
should be fixed by PR #1660
